Add AppArmor package.

sekojima opened the pull request:
# Purpose of pull request

AppArmor 

This pr adds AppArmor package based on meta-security's recipes-mac/AppArmor/apparmor_2.13.2.bb

# Test
## How to test
Change the kernel config to enable AppArmor 
```
CONFIG_TRUSTED_KEYS=y
CONFIG_SECURITY_NETWORK=y
# CONFIG_SECURITY_NETWORK_XFRM is not set
CONFIG_SECURITY_PATH=y
# CONFIG_SECURITY_SELINUX is not set
CONFIG_SECURITY_APPARMOR=y
CONFIG_SECURITY_APPARMOR_BOOTPARAM_VALUE=1
CONFIG_SECURITY_APPARMOR_HASH=y
CONFIG_SECURITY_APPARMOR_HASH_DEFAULT=y
# CONFIG_SECURITY_APPARMOR_DEBUG is not set
CONFIG_DEFAULT_SECURITY_APPARMOR=y
# CONFIG_DEFAULT_SECURITY_DAC is not set
CONFIG_DEFAULT_SECURITY="apparmor"
```
I used this site as a reference.
https://www.kernel.org/doc/html/v4.14/admin-guide/LSM/apparmor.html

Create test script 
Run following command from console
```
# cat <<EOF > test.sh
#!/bin/bash
echo test > sh_write.txt
EOF
# chmod +x test.sh
```

1. Profiles are not loaded on startup and are disabled by default
```
# aa-status
```
2. Create profile
```
# aa-easyprof ~/test.sh > /etc/apparmor.d/home.root.test.sh
# cat /etc/apparmor.d/home.root.test.sh
```
3. Set to enforce mode. Confirm that writing is disabled from the test script.
```
# aa-enforce ~/test.sh
# ./test.sh
# ls
```
4. Set to complain mode. Confirm that writing is enabled from the test script.
```
# aa-complain ~/test.sh
# ./test.sh
# ls
```

## Test result

```
root@qemuarm64:~# aa-status
apparmor module is loaded.
0 profiles are loaded.
0 profiles are in enforce mode.
0 profiles are in complain mode.
0 processes have profiles defined.
0 processes are in enforce mode.
0 processes are in complain mode.
0 processes are unconfined but have a profile defined.
root@qemuarm64:~# aa-easyprof ~/test.sh > /etc/apparmor.d/home.root.test.sh
root@qemuarm64:~# cat /etc/apparmor.d/home.root.test.sh
# vim:syntax=apparmor
# AppArmor policy for test.sh
# ###AUTHOR###
# ###COPYRIGHT###
# ###COMMENT###

#include <tunables/global>

# No template variables specified

"/home/root/test.sh" {
  #include <abstractions/base>

  # No abstractions specified

  # No policy groups specified

  # No read paths specified

  # No write paths specified
}

root@qemuarm64:~# aa-enforce ~/test.sh
Setting /home/root/test.sh to enforce mode.
root@qemuarm64:~# ./test.sh
/bin/bash: ./test.sh: Permission denied
root@qemuarm64:~# ls
test.sh
root@qemuarm64:~# aa-complain ~/test.sh
Setting /home/root/test.sh to complain mode.
root@qemuarm64:~# ./test.sh
root@qemuarm64:~# ls
sh_write.txt  test.sh
```

sekojima committed:
Add AppArmor package.

masami256 commented on recipes-debian/AppArmor/apparmor_debian.bb:
```diff
@@ -0,0 +1,204 @@
+
+require recipes-debian/sources/apparmor.inc
+
+DEBIAN_PATCH_TYPE = "nopatch"
```
debian added some patches (https://sources.debian.org/patches/apparmor/2.13.2-10/). Is there any reasons to not apply them?

sekojima replied:
Fixed to apply debian patch

masami256 commented on recipes-debian/AppArmor/apparmor_debian.bb:
```diff
@@ -0,0 +1,204 @@
+
+PACKAGES += "mod-${PN}"
+
+FILES_${PN} += "/lib/apparmor/ ${sysconfdir}/apparmor ${libdir}/python3/dist-packages /usr/share/" 
```
${datadir} would be nice instead of /usr/share.
FYI: That value is defined in [meta/conf/bitbake.conf](https://git.yoctoproject.org/poky/tree/meta/conf/bitbake.conf?h=warrior)

sekojima replied:
Fix /usr/share to ${datadir}

masami256 commented on recipes-debian/AppArmor/apparmor_debian.bb:
```diff
@@ -0,0 +1,204 @@
+    install -D ${S}/debian/apparmor-easyprof.lintian-overrides ${S}/usr/share/lintian/overrides/apparmor-easyprof
+}
+
+#Building ptest on arm fails.
```
Please describe more information (e.g. "what kind of error you got", "is there any way to fix the error or not",  and so on) in this recipe or commit log.

masami256 replied:
[meta-security's recipe](https://git.yoctoproject.org/meta-security/tree/recipes-mac/AppArmor/apparmor_2.13.2.bb?h=warrior) doesn't have such functions. Does meta-security has same problem?

sekojima replied:
meta-security has the same problem.
Backported the fix commit.
https://git.yoctoproject.org/meta-security/commit/recipes-mac/AppArmor?id=27ddb455543b670097e252ba0d0ad5b7e4101748

masami256 commented on recipes-debian/AppArmor/apparmor_debian.bb:
```diff
@@ -0,0 +1,204 @@
+    oe_runmake -C ${B}/libraries/libapparmor
+}
+
+#Building ptest on arm fails.
```
same as https://github.com/miraclelinux/meta-debian-extended/pull/357/commits/22c4b97dcef5b57b6a8bb9f1de0768916ad5f6c7#r959097545

masami256 commented:
apparmor service doesn't work on sysvinit and systemd environments.

sysvinit

```
root@qemuarm64:~# /etc/init.d/apparmor start
Starting AppArmor profiles:find: /etc/apparmor.d/cache: No such file or directory
/lib/apparmor/functions: line 79: getconf: command not found
xargs: invalid option -- 'd'
BusyBox v1.30.1 (2022-08-31 04:06:05 UTC) multi-call binary.

Usage: xargs [OPTIONS] [PROG ARGS]
/lib/apparmor/functions: line 95: getconf: command not found
xargs: invalid option -- 'd'
BusyBox v1.30.1 (2022-08-31 04:06:05 UTC) multi-call binary.

Usage: xargs [OPTIONS] [PROG ARGS]
 failed!
```

apparmor.service runs /etc/init.d/apparmor script so these failures are same reason.

```
ExecStart=/etc/init.d/apparmor start
ExecStop=/etc/init.d/apparmor stop
ExecReload=/etc/init.d/apparmor reload
```

systemd

```
Aug 31 04:49:01 qemuarm64 systemd[1]: Failed to start AppArmor initialization.
-- Subject: A start job for unit apparmor.service has failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- A start job for unit apparmor.service has finished with a failure.
-- 
-- The job identifier is 100 and the job result is failed.
```

As you can see there is two errors when run /etc/init.d/apparmor.

```
/lib/apparmor/functions: line 79: getconf: command not found
xargs: invalid option -- 'd'
```

The getconf command is provided by glibc-utils package so it need to be listed in RDEPEDNS.
Howevre, busybox's xargs command doesn't support -d option, so you may need to modify to not use -d option or add _findutils_ package in RDEPDNS. findutils is available in /meta/recipes-extended/.

sekojima commented:
> The getconf command is provided by glibc-utils package so it need to be listed in RDEPEDNS.
> Howevre, busybox's xargs command doesn't support -d option, so you may need to modify to not use -d option or add _findutils_ package in RDEPDNS. findutils is available in /meta/recipes-extended/.

Added glibc-utils and findutils packages to RDEPEDNS.

```
root@qemuarm64:~# /etc/init.d/apparmor reload
Reloading AppArmor profiles:find: '/etc/apparmor.d/cache': No such file or directory
find: '/var/cache/apparmor': No such file or directory
find: '/etc/apparmor.d/cache': No such file or directory
[  612.764764] audit: type=1400 audit(1663304487.452:94): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="ping" pid=2699 comm="apparmor_parser"
...snip
[  688.369583] audit: type=1400 audit(1663304563.056:138): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="traceroute" pid=2767 comm="apparmor_parser"
[  690.443146] audit: type=1400 audit(1663304565.128:139): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="winbindd" pid=2769 comm="apparmor_parser"
/lib/apparmor/functions: line 179: comm: command not found
.
```
Added coreutils packages to RDEPEDNS.

sysvinit
```
root@qemuarm64:~# /etc/init.d/apparmor start
Starting AppArmor profiles:find: '/etc/apparmor.d/cache': No such file or directory
[  419.883817] audit: type=1400 audit(1663304294.568:48): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="ping" pid=2596 comm="apparmor_parser"
[  421.037808] audit: type=1400 audit(1663304295.724:49): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nvidia_modprobe" pid=2598 comm="apparmor_parser"
[  421.045785] audit: type=1400 audit(1663304295.732:50): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nvidia_modprobe//kmod" pid=2598 comm="apparmor_parser"
[  421.663115] audit: type=1400 audit(1663304296.344:51): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="klogd" pid=2600 comm="apparmor_parser"
[  423.864491] audit: type=1400 audit(1663304298.552:52): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="syslog-ng" pid=2602 comm="apparmor_parser"
[  425.773125] audit: type=1400 audit(1663304300.460:53): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="syslogd" pid=2604 comm="apparmor_parser"
[  433.497636] audit: type=1400 audit(1663304308.184:54): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2" pid=2606 comm="apparmor_parser"
[  433.502362] audit: type=1400 audit(1663304308.188:55): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2//DEFAULT_URI" pid=2606 comm="apparmor_parser"
[  433.505118] audit: type=1400 audit(1663304308.192:56): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2//HANDLING_UNTRUSTED_INPUT" pid=2606 comm="apparmor_parser"
[  433.507276] audit: type=1400 audit(1663304308.192:57): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2//phpsysinfo" pid=2606 comm="apparmor_parser"
[  434.099740] audit: type=1400 audit(1663304308.788:58): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/anvil" pid=2608 comm="apparmor_parser"
[  436.573626] audit: type=1400 audit(1663304311.260:59): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/auth" pid=2610 comm="apparmor_parser"
[  438.566019] audit: type=1400 audit(1663304313.252:60): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/config" pid=2612 comm="apparmor_parser"
[  440.522794] audit: type=1400 audit(1663304315.208:61): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/deliver" pid=2614 comm="apparmor_parser"
[  442.633049] audit: type=1400 audit(1663304317.320:62): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dict" pid=2616 comm="apparmor_parser"
[  444.897927] audit: type=1400 audit(1663304319.584:63): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dovecot-auth" pid=2618 comm="apparmor_parser"
[  449.147163] audit: type=1400 audit(1663304323.832:64): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dovecot-lda" pid=2620 comm="apparmor_parser"
[  449.153420] audit: type=1400 audit(1663304323.840:65): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dovecot-lda//sendmail" pid=2620 comm="apparmor_parser"
[  451.144365] audit: type=1400 audit(1663304325.832:66): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/imap" pid=2622 comm="apparmor_parser"
[  452.190911] audit: type=1400 audit(1663304326.876:67): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/imap-login" pid=2624 comm="apparmor_parser"
[  454.259774] audit: type=1400 audit(1663304328.948:68): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/lmtp" pid=2626 comm="apparmor_parser"
[  454.833345] audit: type=1400 audit(1663304329.520:69): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/log" pid=2628 comm="apparmor_parser"
[  455.468871] audit: type=1400 audit(1663304330.156:70): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/managesieve" pid=2630 comm="apparmor_parser"
[  456.499320] audit: type=1400 audit(1663304331.184:71): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/managesieve-login" pid=2632 comm="apparmor_parser"
[  458.474788] audit: type=1400 audit(1663304333.160:72): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/pop3" pid=2634 comm="apparmor_parser"
[  460.425450] audit: type=1400 audit(1663304335.112:73): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/pop3-login" pid=2636 comm="apparmor_parser"
[  461.033589] audit: type=1400 audit(1663304335.720:74): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/ssl-params" pid=2638 comm="apparmor_parser"
[  461.611112] audit: type=1400 audit(1663304336.296:75): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/stats" pid=2640 comm="apparmor_parser"
[  469.528001] audit: type=1400 audit(1663304344.216:76): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2" pid=2642 comm="apparmor_parser"
[  469.533782] audit: type=1400 audit(1663304344.220:77): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2//DEFAULT_URI" pid=2642 comm="apparmor_parser"
[  469.535020] audit: type=1400 audit(1663304344.220:78): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2//HANDLING_UNTRUSTED_INPUT" pid=2642 comm="apparmor_parser"
[  469.537940] audit: type=1400 audit(1663304344.224:79): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2//phpsysinfo" pid=2642 comm="apparmor_parser"
[  471.568906] audit: type=1400 audit(1663304346.256:80): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="avahi-daemon" pid=2644 comm="apparmor_parser"
[  474.516547] audit: type=1400 audit(1663304349.204:81): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/sbin/dnsmasq" pid=2646 comm="apparmor_parser"
[  474.519238] audit: type=1400 audit(1663304349.204:82): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/sbin/dnsmasq//libvirt_leaseshelper" pid=2646 comm="apparmor_parser"
[  477.183110] audit: type=1400 audit(1663304351.868:83): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="dovecot" pid=2648 comm="apparmor_parser"
[  479.059025] audit: type=1400 audit(1663304353.744:84): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="identd" pid=2650 comm="apparmor_parser"
[  480.899047] audit: type=1400 audit(1663304355.584:85): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="mdnsd" pid=2652 comm="apparmor_parser"
[  482.943599] audit: type=1400 audit(1663304357.632:86): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nmbd" pid=2654 comm="apparmor_parser"
[  484.971497] audit: type=1400 audit(1663304359.656:87): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nscd" pid=2656 comm="apparmor_parser"
[  487.455665] audit: type=1400 audit(1663304362.144:88): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="ntpd" pid=2658 comm="apparmor_parser"
[  490.078427] audit: type=1400 audit(1663304364.764:89): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="smbd" pid=2660 comm="apparmor_parser"
[  494.094379] audit: type=1400 audit(1663304368.780:90): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="smbldap-useradd" pid=2662 comm="apparmor_parser"
[  494.099824] audit: type=1400 audit(1663304368.788:91): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="smbldap-useradd///etc/init.d/nscd" pid=2662 comm="apparmor_parser"
[  495.990542] audit: type=1400 audit(1663304370.676:92): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="traceroute" pid=2664 comm="apparmor_parser"
[  498.079215] audit: type=1400 audit(1663304372.764:93): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="winbindd" pid=2666 comm="apparmor_parser"
.

root@qemuarm64:~# /etc/init.d/apparmor stop
Clearing AppArmor profiles cache:find: '/etc/apparmor.d/cache': No such file or directory
find: '/var/cache/apparmor': No such file or directory
.
All profile caches have been cleared, but no profiles have been unloaded.
Unloading profiles will leave already running processes permanently
unconfined, which can lead to unexpected situations.

To set a process to complain mode, use the command line tool
'aa-complain'. To really tear down all profiles, run the init script
with the 'teardown' option."

root@qemuarm64:~# /etc/init.d/apparmor reload
Reloading AppArmor profiles:find: '/etc/apparmor.d/cache': No such file or directory
find: '/var/cache/apparmor': No such file or directory
find: '/etc/apparmor.d/cache': No such file or directory
[  690.578347] audit: type=1400 audit(1663306529.124:140): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="ping" pid=2994 comm="apparmor_parser"
[  691.741752] audit: type=1400 audit(1663306530.288:141): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nvidia_modprobe" pid=2996 comm="apparmor_parser"
[  691.746131] audit: type=1400 audit(1663306530.292:142): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nvidia_modprobe//kmod" pid=2996 comm="apparmor_parser"
[  692.367606] audit: type=1400 audit(1663306530.912:143): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="klogd" pid=2998 comm="apparmor_parser"
[  694.607181] audit: type=1400 audit(1663306533.152:144): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="syslog-ng" pid=3000 comm="apparmor_parser"
[  696.527805] audit: type=1400 audit(1663306535.072:145): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="syslogd" pid=3002 comm="apparmor_parser"
[  704.294364] audit: type=1400 audit(1663306542.840:146): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2" pid=3004 comm="apparmor_parser"
[  704.296302] audit: type=1400 audit(1663306542.840:147): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2//DEFAULT_URI" pid=3004 comm="apparmor_parser"
[  704.298861] audit: type=1400 audit(1663306542.844:148): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2//HANDLING_UNTRUSTED_INPUT" pid=3004 comm="apparmor_parser"
[  704.302695] audit: type=1400 audit(1663306542.848:149): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/apache2/mpm-prefork/apache2//phpsysinfo" pid=3004 comm="apparmor_parser"
[  705.023625] audit: type=1400 audit(1663306543.568:150): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/anvil" pid=3006 comm="apparmor_parser"
[  707.579020] audit: type=1400 audit(1663306546.124:151): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/auth" pid=3008 comm="apparmor_parser"
[  709.603035] audit: type=1400 audit(1663306548.148:152): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/config" pid=3010 comm="apparmor_parser"
[  711.578823] audit: type=1400 audit(1663306550.124:153): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/deliver" pid=3012 comm="apparmor_parser"
[  713.591413] audit: type=1400 audit(1663306552.136:154): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dict" pid=3014 comm="apparmor_parser"
[  715.809846] audit: type=1400 audit(1663306554.356:155): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dovecot-auth" pid=3016 comm="apparmor_parser"
[  720.092795] audit: type=1400 audit(1663306558.640:156): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dovecot-lda" pid=3018 comm="apparmor_parser"
[  720.095032] audit: type=1400 audit(1663306558.640:157): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/dovecot-lda//sendmail" pid=3018 comm="apparmor_parser"
[  722.933431] audit: type=1400 audit(1663306561.480:158): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/imap" pid=3020 comm="apparmor_parser"
[  724.506275] audit: type=1400 audit(1663306563.052:159): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/imap-login" pid=3086 comm="apparmor_parser"
[  727.666007] audit: type=1400 audit(1663306566.212:160): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/lmtp" pid=3126 comm="apparmor_parser"
[  728.622150] audit: type=1400 audit(1663306567.168:161): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/log" pid=3200 comm="apparmor_parser"
[  729.583723] audit: type=1400 audit(1663306568.128:162): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/managesieve" pid=3223 comm="apparmor_parser"
[  731.291998] audit: type=1400 audit(1663306569.836:163): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/managesieve-login" pid=3250 comm="apparmor_parser"
[  734.199826] audit: type=1400 audit(1663306572.744:164): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/pop3" pid=3286 comm="apparmor_parser"
[  737.087159] audit: type=1400 audit(1663306575.632:165): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/pop3-login" pid=3360 comm="apparmor_parser"
[  737.759191] audit: type=1400 audit(1663306576.304:166): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/ssl-params" pid=3416 comm="apparmor_parser"
[  738.512011] audit: type=1400 audit(1663306577.056:167): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/lib/dovecot/stats" pid=3418 comm="apparmor_parser"
[  746.573721] audit: type=1400 audit(1663306585.120:168): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2" pid=3420 comm="apparmor_parser"
[  746.578651] audit: type=1400 audit(1663306585.124:169): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2//DEFAULT_URI" pid=3420 comm="apparmor_parser"
[  746.580398] audit: type=1400 audit(1663306585.124:170): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2//HANDLING_UNTRUSTED_INPUT" pid=3420 comm="apparmor_parser"
[  746.582951] audit: type=1400 audit(1663306585.128:171): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="apache2//phpsysinfo" pid=3420 comm="apparmor_parser"
[  748.806521] audit: type=1400 audit(1663306587.352:172): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="avahi-daemon" pid=3422 comm="apparmor_parser"
[  751.777931] audit: type=1400 audit(1663306590.324:173): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/sbin/dnsmasq" pid=3424 comm="apparmor_parser"
[  751.780180] audit: type=1400 audit(1663306590.324:174): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="/usr/sbin/dnsmasq//libvirt_leaseshelper" pid=3424 comm="apparmor_parser"
[  754.466586] audit: type=1400 audit(1663306593.012:175): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="dovecot" pid=3426 comm="apparmor_parser"
[  756.355492] audit: type=1400 audit(1663306594.900:176): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="identd" pid=3428 comm="apparmor_parser"
[  758.210170] audit: type=1400 audit(1663306596.756:177): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="mdnsd" pid=3430 comm="apparmor_parser"
[  760.271148] audit: type=1400 audit(1663306598.816:178): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nmbd" pid=3432 comm="apparmor_parser"
[  762.300808] audit: type=1400 audit(1663306600.848:179): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="nscd" pid=3434 comm="apparmor_parser"
[  764.819684] audit: type=1400 audit(1663306603.364:180): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="ntpd" pid=3436 comm="apparmor_parser"
[  767.523857] audit: type=1400 audit(1663306606.068:181): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="smbd" pid=3438 comm="apparmor_parser"
[  771.556201] audit: type=1400 audit(1663306610.100:182): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="smbldap-useradd" pid=3440 comm="apparmor_parser"
[  771.562307] audit: type=1400 audit(1663306610.108:183): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="smbldap-useradd///etc/init.d/nscd" pid=3440 comm="apparmor_parser"
[  773.472120] audit: type=1400 audit(1663306612.016:184): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="traceroute" pid=3442 comm="apparmor_parser"
[  775.569700] audit: type=1400 audit(1663306614.116:185): apparmor="STATUS" operation="profile_replace" info="same as current profile, skipping" profile="unconfined" name="winbindd" pid=3444 comm="apparmor_parser"
.
```

sekojima commented:
systemd
```
Sep 16 05:56:08 qemuarm64 systemd[1]: Configuration file /lib/systemd/system/apparmor.service is marked executable. Please remove executable permission bits. Proceeding anyway.
Sep 16 05:56:08 qemuarm64 systemd[1]: Condition check resulted in AppArmor initialization being skipped.
-- Subject: A start job for unit apparmor.service has finished successfully
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- A start job for unit apparmor.service has finished successfully.
--
-- The job identifier is 95.
```
apparmor.service is marked executable. 

```
root@qemuarm64:~# ls -la /lib/systemd/system/apparmor.service
-rwxr-xr-x 1 root root 544 Sep 16 05:19 /lib/systemd/system/apparmor.service
```
Removed execute permission bit for apparmor.service.
```
Sep 16 06:52:28 qemuarm64 systemd[1]: Condition check resulted in AppArmor initialization being skipped.
-- Subject: A start job for unit apparmor.service has finished successfully
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- A start job for unit apparmor.service has finished successfully.
--
-- The job identifier is 106.
```

masami256 commented on recipes-debian/AppArmor/apparmor_debian.bb:
```diff
@@ -0,0 +1,208 @@
+do_install_append() {
+    #Install files add by debian package
+    #debian/apparmor.install
+    install -d ${D}/usr/share/apport/package-hooks
```
``` ${D}/usr/share ``` would  be ``` ${D}${datadir} ``` .

sekojima replied:
Fix /usr/share to ${datadir}

masami256 commented:
@sekojima thanks.
